### PR TITLE
GH-35404: [Docs] Fix logo url and temporarily pin sphinx to 5.x to 

### LINKED ISSUE #35404: [Docs] Temporarily pin sphinx to 5.x for the doc build
See https://github.com/apache/arrow/issues/35403 about requiring some fixes on our side to support sphinx 6. Until then, let's pin sphinx to <6.

### LINKED ISSUE #35406: [Website][Docs] Missing logo on Arrow docs page
### Describe the bug, including details regarding any error messages, version, and platform.

There seems to be a missing logo from the generated docs:
https://arrow.apache.org/docs/
image
The logo is visible on other pages, like:
https://arrow.apache.org/docs/format/Versioning.html
image

### Component(s)

Documentation, Website

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
See https://github.com/apache/arrow/issues/35403 about requiring some fixes on our side to support sphinx 6. Until then, let's pin sphinx to <6.

* Closes: #35404
* Closes: #35406

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit preview-docs

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit preview-docs

**Comment by jorisvandenbossche** [comment]
This seems to have worked: http://crossbow.voltrondata.com/pr_docs/35405/

File: docs/source/_templates/docs-sidebar.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 <a class="navbar-brand" href="{{ pathto(master_doc) }}">
-  <img src="{{ pathto('_static/' + logo_url, 1) }}" class="logo" alt="logo">
+  <img src="{{ logo_url|e }}" class="logo" alt="logo">
```
Comment (kou) [inline docs/source/_templates/docs-sidebar.html:3]: Does this work only with Sphinx 5?
If this works with both of Sphinx 5 and 6, how about removing `,<6` for `sphinx` from this PR?

Comment (assignUser) [reply docs/source/_templates/docs-sidebar.html:3]: This works for both but there are other issue (#35403) that require us to pin to <6 for now.

Comment (jorisvandenbossche) [reply docs/source/_templates/docs-sidebar.html:3]: Yes, @assignUser is spot on. This fixes the logo issue (https://github.com/apache/arrow/issues/35406) regardless of the sphinx version, but we still need to fix the jquery dependency issue for being able to use sphinx 6

Comment (kou) [reply docs/source/_templates/docs-sidebar.html:3]: Thanks. I see.

**Pull request merged**